### PR TITLE
failure handling

### PR DESCRIPTION
**Pull request by pavloburchak**
This PR...

## Changes

- added failure handling block and ability to invert test results
- https://github.com/kubeshop/testkube/issues/2482

## Fixes

-

## How to test it

-

## screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/SettingsGeneral.tsx [hunk 1]
```diff
@@ -17,6 +18,7 @@ const SettingsGeneral: React.FC = () => {
       <NameNDescription />
       <Labels />
       {entity === 'tests' ? <Timeout /> : null}
```
Comment (PunkSage) [inline src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/SettingsGeneral.tsx:20]: Do we prefer this convention over equivalent  {entity === 'tests' && <Timeout />}

File: src/components/custom-antd/index.ts [hunk 2]
```diff
@@ -9,3 +9,4 @@ export {default as Upload} from './Upload';
 export {default as Steps} from './Steps';
 export {default as Step} from './Steps/Step';
 export {default as Modal} from './Modal';
+export {default as Checkbox} from './Checkbox';
```
[the remark this answers is not in the document]
Comment (pavloburchak) [reply src/components/custom-antd/index.ts:12]: Exported as usual, how else could it be?
You mean to not export them here and use alias for each?

File: src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/FailureHandling.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        onConfirm={() => {
+          form.submit();
+        }}
+        onCancel={() => {
+          form.resetFields();
+        }}
+      >
+        <StyledFormItem name="negativeTest" valuePropName="checked">
+          <Checkbox>
+            <>
```
[the remark this answers is not in the document]
Comment (pavloburchak) [reply src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/FailureHandling.tsx:75]: No reason indeed

File: src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTest/Source.tsx [hunk 4]
```diff
@@ -134,7 +134,7 @@ const Source = () => {
     })
       .then((res: any) => {
         displayDefaultNotificationFlow(res, () => {
-          notificationCall('passed', `Test source was succesfully updated.`);
+          notificationCall('passed', `Test source was successfully updated.`);
```
[the remark this answers is not in the document]
Comment (pavloburchak) [reply src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTest/Source.tsx:137]: We do not plan, for now

File: src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/SettingsTests.tsx [hunk 5]
```diff
@@ -58,7 +58,7 @@ const SettingsTests = () => {
     })
       .then((res: any) => {
```
[the remark this answers is not in the document]
Comment (pavloburchak) [reply src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/SettingsTests.tsx:59]: We have plan to refactor notifications and overall error handling in future. So for now it is as it is

File: src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/FailureHandling.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+              Invert test result
+              <Popover content={popoverContent}>
+                <QuestionCircleOutlined style={{marginLeft: '10px'}} />
```
[the remark this answers is not in the document]
Comment (pavloburchak) [reply src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/FailureHandling.tsx:78]: There are many places when we have inline styles actually. I don't see anything bad using them, sometimes I find them useful, when adding styled components doesn't make any sense. But if you want to refactor this some day, you are welcome!

File: src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/SettingsGeneral.tsx (hunk 1 above)
Comment (PunkSage) [reply src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsGeneral/SettingsGeneral.tsx:20]: +1

**Review by PunkSage — APPROVED** [review]

**Pull request merged**